src/libraries/Microsoft.PowerFx.Core/Public/Types/ExternalType.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Core.Types;
+
+namespace Microsoft.PowerFx.Core.Public.Types
+{
+    /// <summary>
+    /// This enum represents types which are not supported by PowerFx
+    /// but which may be supported by external data formats.
+    /// </summary>
+    public enum ExternalTypeKind
+    {
+        Array,
```

Comment:
>Array [](http://example.com/codeflow?start=8&length=5)

Can you add some more comments, especially to help clarify why these aren't just FormulaTypes. :

Arrays - PowerFx only supports Single Column Tables. Json supports arrays. 
Object - maybe rename to UntypedObject? (to help distinguish from fx's record).

Reply:
You've got it on the ExternalType class, but it would be good to comment the enum members.

Reply:
Done